src/subdomains/staking/domain/entities/staking.entity.ts

```diff
@@ -83,8 +75,8 @@ export class Staking extends IEntity {
     staking.userId = userId;
     staking.status = StakingStatus.CREATED;
     staking.strategy = strategy;
-    staking.asset = asset;
-    staking.balance = 0;
+
+    staking.balances.push(StakingBalance.create(asset));
```

Comment:
`balances` is undefined at this time. Maybe `staking.balances = [StakingBalance.create(asset)]`?